sw/device/tests/entropy_src_edn_reqs_test.c

```diff
@@ -0,0 +1,223 @@
+  mmio_region_t base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_KEYMGR_BASE_ADDR);
+  CHECK_DIF_OK(dif_keymgr_init(base_addr, kmgr));
+
+  dif_keymgr_config_t config = {.entropy_reseed_interval = 4};
+  CHECK_DIF_OK(dif_keymgr_configure(kmgr, config));
+  CHECK_DIF_OK(dif_keymgr_advance_state(kmgr, NULL));
+}
+
+static void ibex_test(void) {
+  dif_rv_core_ibex_t ibex;
+
+  LOG_INFO("%s", __func__);
+  mmio_region_t base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_core_ibex_init(base_addr, &ibex));
+
+  for (int i = 0; i < 4; i++) {
+    uint32_t rnd;
+    CHECK_DIF_OK(dif_rv_core_ibex_read_rnd_data(&ibex, &rnd));
+  }
+}
+
+static void alert_handler_test(void) {
+  dif_alert_handler_t alert_handler;
+
+  LOG_INFO("%s", __func__);
+  mmio_region_t base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR);
+  CHECK_DIF_OK(dif_alert_handler_init(base_addr, &alert_handler));
+
+  // Enable alert ping fail local alert and configure that to classb.
+  dif_alert_handler_local_alert_t loc_alerts[] = {
+      kDifAlertHandlerLocalAlertAlertPingFail};
+  dif_alert_handler_class_t loc_alert_classes[] = {kDifAlertHandlerClassB};
+
+  dif_alert_handler_escalation_phase_t esc_phases[] = {
+      {.phase = kDifAlertHandlerClassStatePhase0,
+       .signal = 0,
+       .duration_cycles = 2000}};
+
+  dif_alert_handler_class_config_t class_config = {
+      .auto_lock_accumulation_counter = kDifToggleDisabled,
+      .accumulator_threshold = 0,
+      .irq_deadline_cycles = 10000,
+      .escalation_phases = esc_phases,
+      .escalation_phases_len = ARRAYSIZE(esc_phases),
+      .crashdump_escalation_phase = kDifAlertHandlerClassStatePhase1,
+  };
+
+  dif_alert_handler_class_config_t class_configs[] = {class_config,
+                                                      class_config};
+
+  dif_alert_handler_class_t classes[] = {kDifAlertHandlerClassA,
+                                         kDifAlertHandlerClassB};
+  dif_alert_handler_config_t config = {
+      .alerts = NULL,
+      .alert_classes = NULL,
+      .alerts_len = 0,
+      .local_alerts = loc_alerts,
+      .local_alert_classes = loc_alert_classes,
+      .local_alerts_len = ARRAYSIZE(loc_alerts),
+      .classes = classes,
+      .class_configs = class_configs,
+      .classes_len = ARRAYSIZE(class_configs),
+      .ping_timeout = 10,
+  };
+
+  alert_handler_testutils_configure_all(&alert_handler, config,
+                                        kDifToggleEnabled);
+
+  // Trigger the alert handler to escalate.
+  dif_pwrmgr_t pwrmgr;
+  base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR);
+  CHECK_DIF_OK(dif_pwrmgr_init(base_addr, &pwrmgr));
+  dif_pwrmgr_alert_t alert = kDifPwrmgrAlertFatalFault;
+  CHECK_DIF_OK(dif_pwrmgr_alert_force(&pwrmgr, alert));
+}
+
+bool test_main() {
+  dif_aes_t aes;
+  otbn_t otbn;
+  dif_keymgr_t kmgr;
+  entropy_testutils_boot_mode_init();
```

Comment:
We should also run the test in continuous mode and configure the EDN to force a CSRNG trigger a couple of times.

Reply:
Ok to add a TODO linking to an issue if we want to do this later.

Reply:
done